pact/coin-contract/coin.pact

```diff
@@ -183,39 +186,49 @@
   (defun delete-coin (delete-account create-chain-id create-account create-account-guard quantity)
     (with-capability (TRANSFER)
       (debit delete-account quantity)
+
       { "create-chain-id": create-chain-id
       , "create-account": create-account
       , "create-account-guard": create-account-guard
       , "quantity": quantity
-      , "delete-chain-id": (at "chain-id" (chain-data))
+      , "delete-block-height": (at 'block-height (chain-data))
+      , "delete-chain-id": (at 'chain-id (chain-data))
       , "delete-account": delete-account
       , "delete-tx-hash": (tx-hash)
-      }))
+      })
+    )
 
   (defun create-coin (proof)
-    (let ((outputs (at "outputs" (verify-spv "TXOUT" proof))))
-      (enforce (= 1 (length outputs)) "only one tx in outputs")
-      (bind (at 0 outputs)
+    (let ((outputs (at 'outputs (verify-spv "TXOUT" proof))))
+      (bind outputs
         { "create-chain-id":= create-chain-id
         , "create-account" := create-account
         , "create-account-guard" := create-account-guard
         , "quantity" := quantity
         , "delete-tx-hash" := delete-tx-hash
         , "delete-chain-id" := delete-chain-id
         }
-        (enforce (= (at "chain-id" (chain-data)) create-chain-id "enforce correct create chain ID"))
+
+        (enforce
+          (= (at "chain-id" (chain-data))
+            create-chain-id "enforce correct create chain ID"))
+
         (let ((create-id (format "%:%" [delete-tx-hash delete-chain-id])))
           (with-default-read create-id creates-table
             { "exists": false }
             { "exists":= exists }
+
             (enforce (not exists) (format "enforce unique usage of %" [create-id]))
+
             (insert creates-table create-id { "exists": true })
+
             (with-capability (TRANSFER)
               (credit create-account create-account-guard quantity)))
           )))
     )
 
 
+
```

Comment:
Material changes to the Coin Contract are the following:

1. Since we do only ever return 1 object from `verify-spv`, we do not need to consider a list nor enforcing it be a singleton. 

2. Syntactic clean up for legibility reasons.